core/src/main/java/com/xmartlabs/bigbang/core/helper/ui/UiHelper.kt

```diff
@@ -0,0 +1,150 @@
+import android.widget.Spinner
+import com.google.android.gms.common.ConnectionResult
+import com.google.android.gms.common.GoogleApiAvailability
+import java.util.*
+
+object UiHelper {
+  /**
+   * Displays an alert if the Google Play Services are not enabled.
+   *
+   * @param activity the activity to use to check the availability of the Google Play Services
+   * *
+   * @return true if the Google Play Services are available
+   */
+  fun checkGooglePlayServicesAndShowAlertIfNeeded(activity: Activity): Boolean {
+    val googlePlayServicesStatus = GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(activity)
+    
+    if (googlePlayServicesStatus == ConnectionResult.SUCCESS) {
+      return true
+    }
+    
+    if (GoogleApiAvailability.getInstance().isUserResolvableError(googlePlayServicesStatus)) {
+      GoogleApiAvailability.getInstance().getErrorDialog(activity, googlePlayServicesStatus, 0).show()
+    }
+    
+    return false
+  }
+  
+  /**
+   * Retrieves the object in the `adapter` at the `spinner` position.
+   *
+   * @param spinner the spinner to retrieve the position from
+   * *
+   * @param adapter the adapter to retrieve the object from
+   * *
+   * @param <T> the type of the object held by the adapter
+   * *
+   * @return an instance of type T at the spinner position
+  </T> */
+  fun <T> getSpinnerValue(spinner: Spinner, adapter: ArrayAdapter<T>) = adapter.getItem(spinner.selectedItemPosition)
+  
+  /**
+   * Retrieves the URL of the drawable with `drawableResId` id in the given `context`.
+   *
+   * @param context the context from which to retrieve the resource
+   * *
+   * @param drawableResId the id of the drawable resource
+   * *
+   * @return the URL of the drawable
+   */
+  fun getUrlForDrawable(context: Context, @DrawableRes drawableResId: Int) = String.format(
+      Locale.US,
+      "%s://%s/%s/%s",
+      ContentResolver.SCHEME_ANDROID_RESOURCE,
+      context.resources.getResourcePackageName(drawableResId),
+      context.resources.getResourceTypeName(drawableResId),
+      context.resources.getResourceEntryName(drawableResId)
+  )
+  
+  /**
+   * Validates that the field is not empty and, if it's an email field, validates that the email given is valid.
+   * A field is said to be an email field if it's input type is TYPE_TEXT_VARIATION_EMAIL_ADDRESS.
+   *
+   * @param editText the [EditText] instance to validate
+   * *
+   * @return true if the field has content and, in case of an email field, that the content matches a valid email
+   */
+  fun isValidField(editText: EditText) = !editText.text.toString().isEmpty()
+      && (!isEmailField(editText) || isValidEmail(editText.text))
+  
+  /**
+   * Checks whether the given `EditText` is an email field or not.
+   *
+   * @param editText the [EditText] to check
+   * *
+   * @return true if the [EditText] input type is TYPE_TEXT_VARIATION_EMAIL_ADDRESS
+   */
+  fun isEmailField(editText: EditText) = editText.inputType and
```

Comment:
why sometimes using `and` and sometimes `&&`? Maybe we should define a standard :smile: (I love standards!)